packages/auth0_api_python/src/auth0_api_python/api_client.py

```
@@ -14,46 +29,210 @@ class ApiClient:
+            VerifyAccessTokenError: If access token verification fails
+        """
+        authorization_header = headers.get("authorization", "")
+        dpop_proof = headers.get("dpop")
+
+        if not authorization_header:
+            if getattr(self.options, "dpop_required", False):
+                raise self._prepare_error(
+                        InvalidAuthSchemeError("Expecting Authorization header with DPoP scheme.")
+                    )
+            else :
+                raise self._prepare_error(MissingAuthorizationError())
+
+
+        parts = authorization_header.split(" ")
+        if len(parts) != 2:
+            if len(parts) < 2:
+                raise self._prepare_error(MissingAuthorizationError())
+            elif len(parts) > 2:
+                raise self._prepare_error(
+                    InvalidAuthSchemeError("Invalid Authorization HTTP Header Format for authorization")
+                )
+
+
+        try:
+            scheme, token = authorization_header.split(" ", 1)
+        except ValueError:
+            raise self._prepare_error(
+                MissingAuthorizationError()
             )
-        return self._jwks_data
+
+
+        scheme = scheme.strip().lower()
+
+        if getattr(self.options, "dpop_required", False) and scheme != "dpop":
+                if scheme == "bearer":
+                    raise self._prepare_error(
+                        InvalidAuthSchemeError("Invalid scheme. Expected 'DPoP', but got 'bearer'."),
+                        auth_scheme=scheme
+                    )
+                else:
+                    raise self._prepare_error(
+                        InvalidAuthSchemeError("Invalid scheme. Expected 'DPoP' scheme."),
+                        auth_scheme=scheme
+                    )
+        if not token.strip():
+            raise self._prepare_error(MissingAuthorizationError())
+
+
+        if scheme == "dpop":
+            if not self.options.dpop_enabled:
```

Could you please elaborate the reasoning for using `getattr(self.options, "dpop_required", False)` in other places and `self.options.dpop_enabled` here?

---

We intentionally treat these two flags differently to keep our security posture as clear and robust as possible:

`self.options.dpop_enabled` uses direct access because DPoP support is a fundamental capability that must be explicitly enabled or disabled. When disabled, the entire DPoP authentication flow is unavailable, making it a critical security boundary that should fail fast if misconfigured.

`getattr(self.options, "dpop_required", False)` uses defensive programming for an enforcement policy that determines whether DPoP is mandatory. We allow the SDK to default to a safe, “not required” state if that setting isn’t explicitly provided.